### PR TITLE
Avoid forcing "proxied URL" in case of transparent request

### PR DESCRIPTION
**Pull request by ykopel**
In addition to https://github.com/apache/trafficserver/issues/1448
In case of transparent request we don't want to change the URL format that the client sent because we are transparent.

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci]

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/linux-github/1492/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/freebsd-github/1598/ for details.

**Comment by atsci** [comment]
Intel CC build *successful*! See https://ci.trafficserver.apache.org/job/icc-github/30/ for details.

**Comment by zwoop** [comment]
Please run "make clang-format", and push again.

**Comment by atsci** [comment]
clang-analyzer build *successful*! See https://ci.trafficserver.apache.org/job/clang-analyzer-github/162/ for details.

File: proxy/http/HttpTransact.cc [hunk 1]
```diff
@@ -7845,10 +7845,12 @@ HttpTransact::build_request(State *s, HTTPHdr *base_request, HTTPHdr *outgoing_r
       DebugTxn("http_trans", "[build_request] adding target to URL for parent proxy");
       outgoing_request->set_url_target_from_host_field();
     }
-  } else if (s->next_hop_scheme == URL_WKSIDX_HTTP || s->next_hop_scheme == URL_WKSIDX_HTTPS ||
-             s->next_hop_scheme == URL_WKSIDX_WS || s->next_hop_scheme == URL_WKSIDX_WSS) {
+  } else if (!(s->client_info.is_transparent && base_request->is_target_in_url()) &&
```
Comment (SolidWallOfCode) [inline proxy/http/HttpTransact.cc]: Might this not be better as `(! s->client_info.is_transparent && base_request->is_target_in_url())`? If the URL doesn't contain the host name there's no point in checking to see if it needs to be removed.

**Comment by SolidWallOfCode** [comment]
This still needs clang formatting.

File: proxy/http/HttpTransact.cc (hunk 1 above)
Comment (ykopel) [inline proxy/http/HttpTransact.cc]: Its equivalent, it will give the same result.

Comment (ushachar) [inline proxy/http/HttpTransact.cc]: It's not equivalent -- the additional () forces both terms to be evaluated (even if the first one is false)

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/RAT-github/8/ for details.

**Comment by bryancall** [comment]
[approve ci]

**Comment by bryancall** [comment]
@ykopel Can you please run clang-format on it?

File: proxy/http/HttpTransact.cc (hunk 1 above)
Comment (zwoop) [inline proxy/http/HttpTransact.cc]: Feels like we ought to fix that for clarity at least?

**Comment by bryancall** [comment]
If you want to get this in please address the build issues and reopen.

**Pull request closed without merging**